mautic/mautic
# Fix segment multiselect filters

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | ✔️ 
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #4075 #3357
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Segment filters on multiselect fields used to only let in contacts with only one matching option from that multiselect selected. This PR fixes that for contacts with multiple options selected.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
0. **(:star:BONUS:star:)** apply @kuzmany's PR #4080 for proper segment filter display
1. Create a multiselect field ![image](https://cloud.githubusercontent.com/assets/18265735/26241369/b037c72e-3c84-11e7-839d-3fa1a95df4db.png)
2. Create a segment filtering that field ![image](https://cloud.githubusercontent.com/assets/18265735/26241475/1cd192b6-3c85-11e7-8b11-96d9c70b32a8.png)
3. Create a few contacts using that field
    - One with only a single option selected
    - Another with multiple options selected
    - Another with all options selected
4. Run `mautic:segments:update`
5. Only the contacts with only a **single** matching option will get into the segment, those with multiple or all of them will not.

#### Steps to test this PR:
1. Repeat steps 1 to 4 above
2. Check that contacts with multiple or all options selected get in as well

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
